include/openthread.h

```diff
@@ -1895,6 +1895,29 @@ OTAPI ThreadError OTCALL otGetParentInfo(otInstance *aInstance, otRouterInfo *aP
 OTAPI uint8_t OTCALL otGetStableNetworkDataVersion(otInstance *aInstance);
 
 /**
+ * This function pointer is called when an DIAG_GET.rsp is received.
+ *
+ * @param[in]  aMessage      A pointer to the message buffer containing the received DIAG_GET.rsp payload.
+ * @param[in]  aMessageInfo  A pointer to the message info for @p aMessage.
+ * @param[in]  aContext      A pointer to application-specific context.
+ *
+ */
+typedef void (*otReceiveDiagnosticGetCallback)(otMessage aMessage, const otMessageInfo *aMessageInfo,
```

Comment:
For request-response interactions, does it make more sense to include the callback/context as part of the request (i.e. `otSendDiagnosticGet()` in this scenario)?